config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/EncapsulatedAssertionBuilder.java

```diff
@@ -45,7 +45,12 @@ public void buildAssertionElement(Element encapsulatedAssertionElement, PolicyBu
 
     private Encass getEncass(Bundle bundle, String name, AnnotatedBundle annotatedBundle) {
         LOGGER.log(Level.FINE, "Looking for referenced encass: {0}", name);
-        final AtomicReference<Encass> referenceEncass = new AtomicReference<>(bundle.getEncasses().get(name));
+        final AtomicReference<Encass> referenceEncass;
+        if (annotatedBundle != null) {
```

Comment:
Both bundles should have definition for the referenced encass. In case, if the referenced encass is annotated as shared, it might be initialized with some extra details. Hence, we must see it from the annotated bundle.  Am I right?

Reply:
Yes. Annotated bundle stores its own copy of Encass which might have some extra details if marked as shared so we are always looking in the AnnotatedBundle.